### PR TITLE
Fix color and overlap in close dialog header

### LINKED ISSUE #6535: Overlapping close button on Move card dialog heading 0.114.0b0
## Checklist

- [x] I have updated to the latest available Home Assistant version.
- [x] I have cleared the cache of my browser.
- [x] I have tried a different browser to see if it is related to my browser.

## The problem

The "move card" dialog now gained a close button, however this overlaps the heading text.

image

## Steps to reproduce

1. Enter Configure UI mode
2. Select "Move to View" on any card
3. Look at the heading

This can be reproduced on the [demo](https://demo.home-assistant.io/#/lovelace/0).

## Environment

- Home Assistant release with the issue: 0.114.0b0
- Last working Home Assistant release (if known): 0.113
- Browser and browser version: Chrome
- Operating system: macOS

## State of relevant entities

```yaml

```

## Problem-relevant configuration

```yaml

```

## Javascript errors shown in your browser console/inspector

```txt

```

### PR DESCRIPTION
**Pull request by ludeeus**
## Type of change

- [ ] Dependency upgrade
- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #6535
- This PR is related to issue:
- Link to documentation pull request:

### Before

image

### After

image

## Checklist

- [x] The code change is tested and works locally.
- [x] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

File: src/components/ha-dialog.ts [hunk 1]
```diff
@@ -63,16 +63,23 @@ export class HaDialog extends MwcDialog {
...
+          color: var(--text-primary-color);
           position: absolute;
           right: 16px;
           top: 10px;
           text-decoration: none;
-          color: inherit;
+        }
+        .header_title {
+          margin-right: 40px;
+          color: var(--text-primary-color);
```
Comment (bramkragten) [inline src/components/ha-dialog.ts:74]: Why is this needed? We have `--mdc-dialog-heading-ink-color: var(--primary-text-color);
    --mdc-dialog-content-ink-color: var(--primary-text-color);`

Probably the dialog that is not displayed correctly doesn't use `haStyleDialog`

File: src/components/ha-dialog.ts [hunk 2]
```diff
@@ -63,16 +63,23 @@ export class HaDialog extends MwcDialog {
           min-height: var(--mdc-dialog-min-height, auto);
         }
         .header_button {
+          color: var(--text-primary-color);
```
Comment (bramkragten) [inline src/components/ha-dialog.ts:66]: Same.

File: src/components/ha-dialog.ts [hunk 3]
```diff
@@ -63,16 +63,23 @@ export class HaDialog extends MwcDialog {
...
           position: absolute;
           right: 16px;
           top: 10px;
           text-decoration: none;
-          color: inherit;
```
Comment (bramkragten) [inline src/components/ha-dialog.ts:70]: Same.

**Comment by ludeeus** [comment]
You where right, it was that simple

**Review by bramkragten — APPROVED** [review]

**Comment by bramkragten** [comment]
I see you removed the overlap fix, was that intentional? I think that can still happen now?

**Comment by ludeeus** [comment]
Does not happen on this dialog at least anymore

**Comment by bramkragten** [comment]
Yeah because the width was changed by the styles probably, but if the translation is longer it will?

**Comment by ludeeus** [comment]
YUP! :(

**Pull request merged**